### PR TITLE
Refs Task#376116 List bots

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets

- [#376116](https://dev.sun-asterisk.com/issues/376116)

## What's this PR do ?

- [x] Implement list bot screen

## Library
*(List package, library third party add new include version)*

- N/A

## Performance

- [ ] Resolved n + 1 query
- [ ] Run explain query already
- [ ] Time run rake task : 1000 ms

## Checklist

- [ ] It was tested in local success?
- [x] Fill link PR into ticket and the opposite
- [ ] Note purpose/reason, solution, scope of influence into ticket

## ENV note

```
```

## Notes
*(Other notes)*

### CODE REVIEW

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>phpcs</td><td>failure</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>failure</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

File: public/js/tablesDatatables.js [hunk 1]
```diff
@@ -18,6 +18,12 @@ var TablesDatatables = function () {
                 lengthMenu: [[10, 20, 30, -1], [10, 20, 30, 'All']]
             });
 
+            $('#bot-datatable').dataTable({
+                columnDefs: [{orderable: false, targets: [2]}],
+                pageLength: 15,
+                lengthMenu: [[15, 50, 100, -1], [15, 50, 100, 'All']]
```
Comment (tamtt-1031) [inline public/js/tablesDatatables.js]: chắc chỉnh lại spec chỗ này để 5, 10, 20, all thôi e :v

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr></table>

File: app/Models/User.php [hunk 2]
```diff
@@ -40,4 +40,9 @@ class User extends Authenticatable
     protected $casts = [
         'email_verified_at' => 'datetime',
     ];
+
+    public function bots()
+    {
+        return $this->hasMany('App\Models\Bot');
```
Comment (quanghung97) [inline app/Models/User.php]: $this->hasMany('App\Models\Bot');

chuyển thành: $this->hasMany(Bot::class); cho gọn tương tự các models khác

File: app/Http/Controllers/BotController.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Bot;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
```
Comment (quanghung97) [inline app/Http/Controllers/BotController.php]: chúng ta đã register Auth ở config rồi thì chỉ cần 
```
use Auth;
```
là được

File: public/js/app.js [hunk 4]
```diff
@@ -1,113 +1,667 @@
...
+            handleSidebar(mode, extra); // Handle sidebars - access functionality from everywhere
+        },
+        datatables: function() {
+            dtIntegration(); // Datatables Bootstrap integration
+        }
+    };
+}();
+
+/* Initialize app when page loads */
+$(function(){ App.init(); });
```
Comment (ghost) [inline public/js/app.js]: end file

File: app/Http/Controllers/BotController.php [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+class BotController extends Controller
+{
+    public function index()
+    {
+        $bots = Auth::user()->bots;
```
Comment (tamtt-1031) [inline app/Http/Controllers/BotController.php]: ko order theo Thời gian tạo mới nhất à chú?

Comment (ghost) [reply app/Http/Controllers/BotController.php]: cái này bên thằng tabledatatable nó sort theo bên nó, e đang tìm cách remove cái sort của nó đi

File: public/js/tablesDatatables.js (hunk 1 above)
Comment (ghost) [reply public/js/tablesDatatables.js]: khoảng cách nhiều vậy mới tiện hơn a nhỉ

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr></table>

**Review by ghost — APPROVED** [review]

**Comment by framgiaciv3** [comment]
coverage</br> **1. Result build** </br><table><tr><th>Style</th><th>Status</th><th>Ignored</th></tr><tr><td>php_artisan_migrate_._start_run_.13</td><td>success</td><td></td></tr><tr><td>php_artisan_key_gene._start_run_.9</td><td>success</td><td></td></tr><tr><td>framgia-ci_test-conn._start_run_.11</td><td>success</td><td></td></tr><tr><td>composer_install._start_run_.5</td><td>success</td><td></td></tr><tr><td>phpcs</td><td>success</td><td></td></tr><tr><td>phpunit</td><td>success</td><td></td></tr><tr><td>cp__env_example__env._start_run_.3</td><td>success</td><td></td></tr><tr><td>php_artisan_config_c._start_run_.7</td><td>success</td><td></td></tr></table>

**Review by quanghung97 — APPROVED** [review]
ok

**Review by nguyenthinhhn — APPROVED** [review]

**Review by tamtt-1031 — APPROVED** [review]

**Comment by tamtt-1031** [comment]
lgtm :+1:

**Comment by phucqd-1305** [comment]
OK

**Pull request merged**